### PR TITLE
Initial submission

### PR DESCRIPTION
**Pull request by jacob-berger**
Cannot get test to show up, so is untested as of yet.

### CODE REVIEW

File: DelegateFun/SorterTests/SortUtilityTests.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
```
Comment (mmwoodfo) [inline DelegateFun/SorterTests/SortUtilityTests.cs]:
```suggestion

```

File: DelegateFun/SorterTests/SortUtilityTests.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+
+namespace Sorter.Tests
+{
+    class SortUtilityTests
```
Comment (mmwoodfo) [inline DelegateFun/SorterTests/SortUtilityTests.cs]:
```suggestion
    [TestClass()]
    public class SortUtilityTests
```

File: DelegateFun/SorterTests/SortUtilityTests.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        [TestMethod]
+        public void SortUtilityTestAscending()
+        {
+            int[] items = new int[] { 2, 4, 2, 3, 7, 9, 2, 3 };
+            string expected = "";
+            foreach (int item in items)
+            {
+                expected += item + ",";
+            }
```
Comment (mmwoodfo) [inline DelegateFun/SorterTests/SortUtilityTests.cs]: Remove this

File: DelegateFun/SorterTests/SortUtilityTests.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace Sorter.Tests
+{
+    class SortUtilityTests
+    {
+        [TestMethod]
+        public void SortUtilityTestAscending()
+        {
+            int[] items = new int[] { 2, 4, 2, 3, 7, 9, 2, 3 };
+            string expected = "";
```
Comment (mmwoodfo) [inline DelegateFun/SorterTests/SortUtilityTests.cs]:
```suggestion
            int[] expected = { 2, 2, 2, 3, 3, 4, 7, 9 };
```

File: DelegateFun/SorterTests/SortUtilityTests.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            SortUtility.Sort(items, delegate(int first, int second)
+            {
+                return first > second;
+            });
+
+            string result = "";
+            foreach (int item in items)
+            {
+                result += item + ",";
+            }
```
Comment (mmwoodfo) [inline DelegateFun/SorterTests/SortUtilityTests.cs]: Remove this

File: DelegateFun/SorterTests/SortUtilityTests.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+            Assert.AreEqual(expected, result);
```
Comment (mmwoodfo) [inline DelegateFun/SorterTests/SortUtilityTests.cs]:
```suggestion
            CollectionAssert.AreEqual(expected, items);
```

File: DelegateFun/SorterTests/SortUtilityTests.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            string expected = "";
+            foreach (int item in items)
+            {
+                expected += item + ",";
+            }
+
+
+            SortUtility.Sort(items, delegate(int first, int second)
+            {
+                return first > second;
```
Comment (mmwoodfo) [inline DelegateFun/SorterTests/SortUtilityTests.cs]: If you are going to sort in ascending order flip the sign
```suggestion
                return first < second;
```

File: DelegateFun/SorterTests/SortUtilityTests.cs [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+            //Assert
+            for (int ix = 0; ix < items.Length - 1; ix++)
+            {
+                Assert.IsTrue(items[ix] <= items[ix + 1]);
+            }
+        }
+
+        [TestMethod]
+        public void SortAscendingLambdaStatement()
```
Comment (breaman) [inline DelegateFun/SorterTests/SortUtilityTests.cs]: The assignment was suppose to have a different sort algorithm per method type.

**Pull request closed without merging**